app/routes/organization/settings.js

```diff
@@ -1,7 +1,22 @@
-import Route from '@ember/routing/route';
+import TravisRoute from 'travis/routes/basic';
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
+
+export default TravisRoute.extend({
+  features: service(),
 
-export default Route.extend({
   beforeModel() {
-    this.transitionTo('organization.repositories', this.modelFor('organization'));
-  }
+    if (!this.get('features.proVersion')) {
+      this.transitionTo('organization.repositories', this.modelFor('organization'));
+    }
+  },
+
+  model() {
+    const organization = this.modelFor('organization');
+    if (organization.permissions.admin !== true) {
+      this.transitionTo('organization.repositories', organization);
+    }
+    const preferences = this.store.query('preference', { organization_id: organization.id });
```

Comment:
It seems like `preferences` is a good case for `organization.hasMany('preference', { async: true })`, WDYT? It would allow for better request/caching handling and proper payload extraction (e.g. camelize property keys)

Reply:
Yeah that seems like maybe the better way to do it. I haven't worked a lot with ember-data relationships yet, what would be involved?

Reply:
In general, first I would create a model (as soon as we already have `preference` model for user preferences, I would give it a name like `orgPreference`). Then set it as a relation on `organization` model: `preferences: hasMany('orgPreference', { async: true })`. Then on the route model I would try to call it:
```js
model() {
  return this.modelFor('organization').get('preferences'); // async relationships return promises
}
```
and see how it goes.
Apparently, in this case it would fail, because it would try to fetch `/orgPreferences` endpoint, so we have to tweak adapter: generate an adapter for `orgPreference`, make it extend `V3Adapter`, add `pathForType` hook and return the correct url, override other hooks (described in the EmberData docs for JSONAPIAdapter) to make sure, that it makes proper requests. Then I would take a look how it extracts data - if it looks good, then just leave it as is, otherwise, do the same as for Adapter but for Serializer.

Although, since the release is planned for today, probably it would make sense to leave it as it is for now and maybe approach it sometime later.

Reply:
Thanks for all this info, it's super helpful! The release got pushed back to tomorrow, so if I end up with time for it I'll try to fit this in.

Reply:
Ok I think given where things are at right now, I don't think there will be time to do this before launch. So yeah I think for now let's leave this as is. I'm already planning on writing up a few issues after the launch, so maybe I'll write one up for this as well.